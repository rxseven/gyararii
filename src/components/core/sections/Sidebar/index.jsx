import React from 'react';

import withContext from 'HOCs/withContext';

function Sidebar(props) {
  return <div>Sidebar</div>;
}

export default withContext(Sidebar);
