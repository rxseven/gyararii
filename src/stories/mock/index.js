import elements from './elements';

export default {
  elements
};
