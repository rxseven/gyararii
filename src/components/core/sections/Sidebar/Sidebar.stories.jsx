import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Sidebar from './index';

const { content } = mock.elements;
const title = 'Components/Core/Sections/Sidebar';
const component = <Sidebar>{content}</Sidebar>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
