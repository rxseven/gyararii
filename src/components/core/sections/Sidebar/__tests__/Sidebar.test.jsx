import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Environment from 'tests/environment';
import mock from 'tests/mock';
import Sidebar from '../index';

const seed = {
  props: {
    children: mock.elements.children
  }
};

describe('<Sidebar />', () => {
  // Arrange
  const component = (
    <Environment>
      <Sidebar>{seed.props.children}</Sidebar>
    </Environment>
  );

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
