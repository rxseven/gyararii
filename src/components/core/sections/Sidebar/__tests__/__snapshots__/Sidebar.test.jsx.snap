// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Snapshot tests should render correctly 1`] = `
<div>
  <div
    className="bm-overlay"
    onClick={[Function]}
    style={
      Object {
        "MozTransform": "translate3d(100%, 0, 0)",
        "MsTransform": "translate3d(100%, 0, 0)",
        "OTransform": "translate3d(100%, 0, 0)",
        "WebkitTransform": "translate3d(100%, 0, 0)",
        "background": "rgba(0, 0, 0, 0.8)",
        "height": "100%",
        "opacity": 0,
        "position": "fixed",
        "top": 0,
        "transform": "translate3d(100%, 0, 0)",
        "transition": "opacity 0.3s, transform 0s 0.3s",
        "width": "100%",
        "zIndex": 1000,
      }
    }
  />
  <div
    className="bm-menu-wrap"
    id="sidebar"
    style={
      Object {
        "MozTransform": "translate3d(-100%, 0, 0)",
        "MsTransform": "translate3d(-100%, 0, 0)",
        "OTransform": "translate3d(-100%, 0, 0)",
        "WebkitTransform": "translate3d(-100%, 0, 0)",
        "height": "100%",
        "maxWidth": "320px",
        "position": "fixed",
        "right": "inherit",
        "top": 0,
        "transform": "translate3d(-100%, 0, 0)",
        "transition": "all 0.5s",
        "width": "85%",
        "zIndex": 8001,
      }
    }
  >
    <div
      className="bm-menu"
      style={
        Object {
          "background": "#fff",
          "boxSizing": "border-box",
          "height": "100%",
          "overflow": "auto",
        }
      }
    >
      <nav
        className="bm-item-list"
        style={
          Object {
            "height": "100%",
            "padding": 0,
            "position": "relative",
          }
        }
      >
        <div
          className="bm-item sc-htpNat bcNAOR"
          style={
            Object {
              "display": "block",
            }
          }
          tabIndex={-1}
        >
          <div
            className="sc-bxivhb qapvx"
          >
            <a
              className="sc-bdVaJa jQwNdM"
              href="/"
              onClick={[Function]}
              title="Gallery"
            >
              ギャラリー
            </a>
          </div>
          <ul
            className="sc-bZQynM hRyDTQ"
          >
            <li
              className="sc-gzVnrw kXANwb"
            >
              <a
                aria-current="page"
                className="sc-htoDjs ydPJF sc-bwzfXH dRlolo active"
                data-light={false}
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-home fa-w-18 "
                  data-icon="home"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="sc-dnqmqq cpJVKj"
                >
                  Gallery
                </span>
              </a>
            </li>
            <li
              className="sc-gzVnrw kXANwb"
            >
              <a
                aria-current={null}
                className="sc-htoDjs ydPJF sc-bwzfXH dRlolo"
                data-light={false}
                href="/about"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-info-circle fa-w-16 "
                  data-icon="info-circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="sc-dnqmqq cpJVKj"
                >
                  About Gyararii
                </span>
              </a>
            </li>
            <li
              className="sc-gzVnrw kXANwb"
            >
              <a
                aria-current={null}
                className="sc-htoDjs ydPJF sc-bwzfXH dRlolo"
                data-light={false}
                href="/terms"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-file-alt fa-w-12 "
                  data-icon="file-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 384 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm64 236c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-64c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-72v8c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12zm96-114.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="sc-dnqmqq cpJVKj"
                >
                  Terms of Use
                </span>
              </a>
            </li>
            <li
              className="sc-gzVnrw kXANwb"
            >
              <a
                aria-current={null}
                className="sc-htoDjs ydPJF sc-bwzfXH dRlolo"
                data-light={false}
                href="/404"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-exclamation-triangle fa-w-18 "
                  data-icon="exclamation-triangle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="sc-dnqmqq cpJVKj"
                >
                  404
                </span>
              </a>
            </li>
            <li
              className="sc-gzVnrw kXANwb"
            >
              <a
                className="sc-ifAKCX jSbHxJ"
                href="https://github.com/rxseven"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-folder-open fa-w-18 "
                  data-icon="folder-open"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M527.9 224H480v-48c0-26.5-21.5-48-48-48H272l-64-64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h400c16.5 0 31.9-8.5 40.7-22.6l79.9-128c20-31.9-3-73.4-40.7-73.4zM48 118c0-3.3 2.7-6 6-6h134.1l64 64H426c3.3 0 6 2.7 6 6v42H152c-16.8 0-32.4 8.8-41.1 23.2L48 351.4zm400 282H72l77.2-128H528z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="sc-dnqmqq cpJVKj"
                >
                  Storybook
                </span>
              </a>
            </li>
            <li
              className="sc-gzVnrw kXANwb"
            >
              <a
                className="sc-ifAKCX jSbHxJ"
                href="https://github.com/rxseven/gyararii"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-github-alt fa-w-15 "
                  data-icon="github-alt"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 480 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="sc-dnqmqq cpJVKj"
                >
                  View on GitHub
                </span>
              </a>
            </li>
          </ul>
          <div
            className="sc-EHOje kTJQym"
          >
            <a
              className="sc-iwsKbI ktpWoE"
              href="https://github.com/rxseven/gyararii/releases/tag/v0.1.0"
              rel="noopener noreferrer"
              target="_blank"
            >
              Version 0.1.0
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
